Sources/Commands/SwiftTool.swift

```diff
@@ -645,13 +629,18 @@ public class SwiftTool {
             // with a map of the names of tools available to each plugin. In particular this would not work with any
             // binary targets.
             let builtToolsDir = dataDir.appending(components: try self._hostToolchain.get().triple.tripleString, buildEnvironment.configuration.dirname)
-            let diagnostics = DiagnosticsEngine()
             
             // Create the cache directory, if needed.
             try localFileSystem.createDirectory(cacheDir, recursive: true)
 
             // Ask the graph to invoke plugins, and return the result.
-            let result = try graph.invokePlugins(outputDir: outputDir, builtToolsDir: builtToolsDir, pluginScriptRunner: pluginScriptRunner, diagnostics: diagnostics, fileSystem: localFileSystem)
+            let result = try graph.invokePlugins(
+                outputDir: outputDir,
+                builtToolsDir: builtToolsDir,
+                pluginScriptRunner: pluginScriptRunner,
+                diagnostics: ObservabilitySystem.makeDiagnosticsEngine(),
```

Comment:
This is an example of where we'd want some kind of chained context, I think.  We're invoking plugins on behalf of a particular build, but then there will be subcontext for each target, and then for each plugin under that.  So that debug output or runtime diagnostics from each plugin has a chained scope of context.  I know that that will be a future PR, but that part of the structural cleanup will really open up a way to show the diagnostics in a much richer way in an IDE.  Not a comment on this PR but forwardlooking.

Reply:
yes exactly, there is an example of doing that in the one module I did convert, basically we can forward the DiagnosticsContext so that we emit in the right context